src/components/spectrogram/spectrogram.ts

```diff
@@ -114,33 +114,41 @@ export class Spectrogram extends SignalWatcher(AbstractComponent(LitElement)) {
     OeResizeObserver.instance.unobserve(this.canvas);
   }
 
-  public willUpdate(change: PropertyValues<this>): void {
-    if (change.has("paused")) {
-      this.setPlaying();
-    }
-
-    // if the src changes, we want to start the recording from the beginning
-    if (change.has("src") || change.has("slotElements")) {
-      this.currentTime.value = 0;
-    }
-  }
-
   public updated(change: PropertyValues<this>) {
-    this.setPlaying();
-
     if (change.has("offset") || change.has("renderWindow")) {
       this.shadowRoot?.dispatchEvent(new Event("slotchange"));
     }
 
-    // spectrogram regeneration functionality
-    if (this.doneFirstRender && this.shouldInvalidateSpectrogram(change)) {
-      this.audioHelper.regenerateSpectrogram(this.mediaElement, this.spectrogramOptions());
+    if (this.doneFirstRender) {
+      // spectrogram regeneration functionality
+      if (this.invalidateSpectrogramOptions(change)) {
+        this.audioHelper.regenerateSpectrogramOptions(this.spectrogramOptions());
+      }
+
+      if (this.invalidateSpectrogramSource(change)) {
+        this.currentTime.value = 0;
+        this.regenerateSpectrogram();
+      }
```

Comment:
probably want to make sure we dont do both (else if)

Reply:
Yeah, I found that this caused a bug with re-rendering src attributes and have fixed it in an upcoming commit